marimo/_ast/codegen.py

```diff
@@ -174,6 +175,12 @@ def generate_filecontents(
 
     unshadowed_builtins = set(builtins.__dict__.keys()) - defs
     fndefs: list[str] = []
+
+    # Update default names to be __{4char_hash}
```

Comment:
A comment like this could go stale. Instead:

```suggestion
    # Canonicalize internal names to the default cell name based on code
```